### PR TITLE
feat(plugins): Create `concurrency` pytest plugin

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
Creates a new plugin that facilitates concurrency in all test-related commands such as fill, consume and execute.

Before the plugin, all different plugins created its own temporary folder in order to save information that was meant to be share across different `xdist` processes (e.g. the execution client running the tests in consume, or in execute hive), and it relied on each plugin having the temporary folder being passed by the command caller for all instances to coincide on the same folder name.

Now the concurrency plugin uses testrun_uid to generate a deterministic temporary folder that will be the same across processes, and can be used for coordination.

## 🔗 Related Issues
Functionality initially added in #712.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: src/pytest_plugins/concurrency.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+"""
+Pytest plugin to create a temporary folder for the session where multi-process tests can store data
+that is shared between processes.
```
Comment (danceratopz) [inline src/pytest_plugins/concurrency.py:3]:
```suggestion
Pytest plugin to create a temporary folder for the session where 
multi-process tests can store data that is shared between processes.

The provided `session_temp_folder` fixture is used, for example, by `consume` 
when running hive simulators to ensure that only one `test_suite` is created
(used to create tests on the hive simulator) when they are being created using
multiple workers with pytest-xdist.
```

File: src/pytest_plugins/concurrency.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from filelock import FileLock
+
+
+@pytest.fixture(scope="session")
+def session_temp_folder_name(testrun_uid: str) -> str:  # noqa: SC200
+    """
+    Define the name of the temporary folder that will be shared among all the
+    xdist workers to coordinate the tests.
+
+    "testrun_uid" is a fixture provided by the xdist plugin, and is unique for each test run,
```
Comment (danceratopz) [inline src/pytest_plugins/concurrency.py:22]: Nice!

**Review by danceratopz — APPROVED** [review]
Really nice abstraction. So nice to get this out of the cli and abstracted to this plugin.

LGTM, just a few minor comments, mainly help/docstrings below.

**Comment by marioevz** [comment]
@danceratopz awesome suggestions, thanks for the review, I'll merge in a bit :)

**Pull request merged**